subsys/net/ip/ipv4_fragment.c

```diff
@@ -467,7 +467,7 @@ static int send_ipv4_fragment(struct net_pkt *pkt, uint16_t rand_id, uint16_t fi
 
 	ipv4_hdr = (struct net_ipv4_hdr *)net_pkt_get_data(frag_pkt, &ipv4_access);
 	if (!ipv4_hdr) {
-		return -ENOBUFS;
+		goto fail;
```

Comment:
I think we need to set the return value here, so
```
ret = -ENOBUFS;
goto fail;
```

Reply:
the return value is initialized to -ENOBUFS in the function entry.

Reply:
Indeed, good point, I did not looked the whole function.